docs/machine-learning/how-to-guides/load-data-ml-net.md

```diff
@@ -0,0 +1,135 @@
+    [VectorType(3)]
+    public float[] HistoricalPrices { get; set; }
+    [LoadColumn(4)]
+    public float CurrentPrice { get; set; }
+}
+```
+
+### Column Attributes
+
+ML.NET provides additional ways to extend data models through attributes.
+
+The [`LoadColumn`](xref:Microsoft.ML.Data.LoadColumnAttribute) attribute specifies your properties' column indices.
+
+Load columns as: 
+- Individual columns like `Size` and `CurrentPrices` in the previous `HousingData` class example.
+- Multiple columns at a time in the form of a vector like `HistoricalPrices` in the previous `HousingData` class example.
+
+If you have a vector property, use the [`VectorType`](xref:Microsoft.ML.Data.VectorTypeAttribute) attribute in your model. One thing to note however is that all of the elements in the vector need to be of the same type.
+
+## Load Data From A File
+
+To load data from a file use the [`LoadFromTextFile`](xref:Microsoft.ML.TextLoaderSaverCatalog.LoadFromTextFile*) method along with the data model schema for the data to be loaded. Since `separatorChar` parameter is tab-delimited by default, change it for your data file as needed. If your file has a header, set the `hasHeader` parameter to `true`.
+
+```csharp
+//Create MLContext
+MLContext mlContext = new MLContext();
+
+//Load Data
+IDataView data = mlContext.Data.LoadFromTextFile<HousingData>("my-data-file.csv", separatorChar: ',', hasHeader: true);
+```
+
+## Load Data From Multiple Files
+
+In the event that your data is stored in multiple files, as long as the schema is the same, ML.NET allows you to load data from multiple files that are either in the same directory or multiple directories.
+
+### Single Directory
+
+When all of your data files are in the same directory, use wildcards in the [`LoadFromTextFile`](xref:Microsoft.ML.TextLoaderSaverCatalog.LoadFromTextFile*) method.
+
+```csharp
+//Create MLContext
+MLContext mlContext = new MLContext();
+
+//Load Data File
+IDataView data = mlContext.Data.LoadFromTextFile<HousingData>("Data/*", separatorChar: ',', hasHeader: true);
+```
+
+### Multiple Directories
+
+To load data from multiple directories, use the [`CreateTextLoader`](xref:Microsoft.ML.TextLoaderSaverCatalog.CreateTextLoader*) method to create a [`TextLoader`](xref:Microsoft.ML.Data.TextLoader). Then, use the [`TextLoader.Load`](xref:Microsoft.ML.DataLoaderExtensions.Load*) method and specify the individual file paths (wildcards can't be used).
+
+```csharp
+//Create MLContext
+MLContext mlContext = new MLContext();
+
+// Create Data Loader
+TextLoader textLoader = mlContext.Data.CreateTextLoader<HousingData>(separatorChar: ',', hasHeader: true);
+
+// Load Data
+IDataView data = textLoader.Load("DataFolder/SubFolder1/1.txt", "DataFolder/SubFolder2/1.txt");
+```
+
+## Load Data From A Streaming Source
+
+Loading data in ML.NET is not limited to data stored on disk. Load data from streaming sources that include but are not limited to:
+
+- In-Memory Collections
```

Comment:
I think it would be worth saying that to load from any of these sources, you need to load into an IEnumerable first